### PR TITLE
support memory mapping

### PR DESCRIPTION
**Pull request by karasikov**
Added support for loading `sdsl::int_vector<*>` with memory mapping. So practically all bit/int-vectors and other data structures in metagraph become memory mapped, including the main graph representations as well as almost all annotation representations.

OFF by default. To turn on, add the `--mmap` flag.

### CODE REVIEW

**Comment by hmusta** [comment]
Some parts that could still use mmap:
- [ ] ColumnCompressed: *load* methods
- [ ] RowCompressed: *load* methods
- [ ] StaticBinRelAnnotator: merge_load_row_diff, load_coords
- [ ] Elias Fano methods?
- [ ] RowDiff: load_* and builder
- [ ] Chunk-based graph builders?
- [ ] Node weights
- [ ] BOSS: load(std::string) (do we still use this?), chunk loading
- [ ] BRWT builders

File: metagraph/src/graph/representation/succinct/dbg_succinct.cpp [hunk 1]
```diff
@@ -648,14 +649,15 @@ bool DBGSuccinct::load_without_mask(const std::string &filename) {
...
-        if (!boss_graph_->load(instream))
+        if (!boss_graph_->load(*in))
             return false;
 
-        mode_ = static_cast<Mode>(load_number(instream));
+        mode_ = static_cast<Mode>(load_number(*in));
 
-        if (!boss_graph_->load_suffix_ranges(instream))
+        if (!boss_graph_->load_suffix_ranges(*in))
             logger->warn("No index for node ranges could be loaded");
```
Comment (hmusta) [inline metagraph/src/graph/representation/succinct/dbg_succinct.cpp:661]: Could you please double-check this? I'm noticing that this warning is being printed for all graphs

**Review by hmusta — CHANGES_REQUESTED** [review]

File: metagraph/src/graph/representation/succinct/dbg_succinct.cpp (hunk 1 above)
Comment (karasikov) [reply metagraph/src/graph/representation/succinct/dbg_succinct.cpp:661]: This is a different warning...

Comment (hmusta) [reply metagraph/src/graph/representation/succinct/dbg_succinct.cpp:661]: false alarm

**Pull request merged**